### DISCUSSION TITLE
Using AIRCFT data type as a profile

### DISCUSSION THREAD

**Discussion by PerryShafran-NOAA** [dtcenter/METplus#1860] [Configuration] [answered]
Hi, there,

We have prepbufr files that we have profiles of aircraft data, but it would appear that METplus isn't set up to do aircraft profiles - only ADPUPA?  Would you be able to help us set it up so we can use these prepbufr files as profiles?  I would be happy to send you an example file.

Thanks!

Perry

**Comment by JohnHalleyGotway** [comment]
@PerryShafran-NOAA I see you have a question about verifying against aircraft point observations. I believe that these are stored as the AIRCFT message type in those files. Configuring METplus to extract and verify against that message type shouldn't be too difficult. If you get stuck along the way, please follow up with specific questions.

To demonstrate, I ran PB2NC to extract `message_type = [ "AIRCFT" ];` and plotted the result. Not all that exciting, but there's definitely some data:
```
DEBUG 2: Total Messages processed		= 70884
DEBUG 2: Rejected based on message type	= 69698
...
DEBUG 2: Total Messages retained		= 1186
DEBUG 2: Total observations retained or derived	= 2802
```


![Screen Shot 2022-10-10 at 3 40 40 PM](https://user-images.githubusercontent.com/21087144/194956519-d26bb006-b826-441b-8b40-c1e88d91e2de.png)

**Comment by PerryShafran-NOAA** [comment]
Hi, @JohnHalleyGotway ,

Specific point observations are easy, but we have specific files that create profiles of aircraft observations which you may or may not be aware of.  And thus, using those specific files, I tried to validate PBL, for example, against these aircraft profiles, but did not get any data in the resulting stat file.  You may have PBL and CAPE calculations specifically set for ADPUPA only, and if we could use AIRCFT for such verification, that would be important to us.

Thanks!

Perry

**Comment by TaraJensen** [comment]
@PerryShafran-NOAA @JohnHalleyGotway - this work sounds pretty much like the type of work @DanielAdriaansen at NCAR is doing for the NRL and @jason-english is doing at GSL.  Sounds like it would be good to have a joint meeting to make sure we are addressing all needs. @DanielAdriaansen @jason-english - please comment here if you'd like.

**Reply by DanielAdriaansen** [reply]
I am happy to have a joint meeting once I get a little farther along- I am using research aircraft data but @jason-english may be a better analog for what Perry is trying to do. I vaguely recall Jason mentioning something about AMDAR or ACARS data, which can be profiles of data as aircraft ascend or descend. Should we facilitate the meeting @TaraJensen ?

**Reply by jason-english** [reply]
Hi @PerryShafran-NOAA, How are you calculating PBL height from the aircraft data? (it has to be derived from existing variables such as temperature and pressure, correct?). I'm (just starting) to utilize AMDAR aircraft data to calculate/estimate PBL height and compare to HRRR model output of PBLH. I haven't decided yet which MET/METplus tools to use, but am looking at using temperature, pressure, and ascent/descent data to generate the PBL. Although we have some differences in our work (I'm using model output while you are looking at PREPBUFR), I think there's a lot of overlap and a joint meeting could be useful; possibly with the two of us, @DanielAdriaansen, @JohnHalleyGotway, and/or others.

**Comment by PerryShafran-NOAA** [comment]
I need to provide y'all with a sample of our prepbufr files that contain profiles of aircraft data.  These profiles are created by taking the individual aircraft observations that are ascents or descents, and then appending the nearest airport station surface data.  Thus we have profiles that look similar to the ADPUPA datasets.  And then they should be calculated in the exact way that we calculate PBL from ADPUPA profiles.

**Reply by JohnHalleyGotway** [reply]
Perry, just catching up on (very) old discussions. Looks like we actually need to create a MET issue to enhance PB2NC to derive PBL from aircraft "soundings". Is that correct? I'll add this to the METplus NOAA discussion for next Monday.

**Reply by JohnHalleyGotway** [reply, marked as answer]
Please see dtcenter/MET#2503 which was created to capture this feature request. I'll mark this discussion as being answered and lock it to prevent future posts. Please feel free to comment directly on dtcenter/MET#2503 with any additional comments or questions.

**Comment by jason-english** [comment]
I'm working on at METplus / METexpress use case where we read in the hourly AMDAR files to generate soundings and then compute PBLH from those soundings. I'm comparing to operational HRRR and RRFS output, so I'm not sure the best way to work with prepbufr files, but there's probably some synergy here.

**Reply by DanielAdriaansen** [reply]
> I'm working on at METplus / METexpress use case where we read in the hourly AMDAR files to generate soundings and then compute PBLH from those soundings. I'm comparing to operational HRRR and RRFS output, so I'm not sure the best way to work with prepbufr files, but there's probably some synergy here.

Just popping in to let you know that your effort was mentioned and communicated to Perry and others at the meeting @jason-english
